time_advance.m

```diff
@@ -66,11 +66,17 @@
     %     k3 = invMatLHS * (k3 + rhs3);
     k3 = ALHS \ (k3 + rhs3);
 else
-    k1 = apply_A(pde,opts,A_data,f,deg,Vmax,Emax)   + source1 + bc1;
-    y1 = f + dt*a21*k1;
-    k2 = apply_A(pde,opts,A_data,y1,deg,Vmax,Emax) + source2 + bc2;
-    y2 = f + dt*(a31*k1 + a32*k2);
-    k3 = apply_A(pde,opts,A_data,y2,deg,Vmax,Emax) + source3 + bc3;
+     k1 = apply_A(pde,opts,A_data,f,deg,Vmax,Emax)   + source1 + bc1;
+     y1 = f + dt*a21*k1;
+     k2 = apply_A(pde,opts,A_data,y1,deg,Vmax,Emax) + source2 + bc2;
+     y2 = f + dt*(a31*k1 + a32*k2);
+     k3 = apply_A(pde,opts,A_data,y2,deg,Vmax,Emax) + source3 + bc3;
+
+%    k1 = apply_A(pde,opts,A_data,f,deg,Vmax,Emax)   + source1 - bc1;
+%    y1 = f + dt*a21*k1;
+%    k2 = apply_A(pde,opts,A_data,y1,deg,Vmax,Emax) + source2 - bc2;
+%    y2 = f + dt*(a31*k1 + a32*k2);
+%    k3 = apply_A(pde,opts,A_data,y2,deg,Vmax,Emax) + source3 - bc3;
```

Comment:
This looks like we need to uncomment the `-bc` section and remove the `+bc` section? (looks like this has not been updated based on what we discovered this morning, i.e., we need `-bc` in all time advance schemes)

Reply:
As discussed earlier, the +bc's are the necessary lines of code in Runge_Kutta that yields successful results for advection1 and advection1reverse.